pyblazing/pyblazing/apiv2/context.py

```diff
@@ -145,26 +154,27 @@ def initializeBlazing(ralId=0, networkInterface='lo', singleNode=False,
 
     return ralCommunicationPort, workerIp, cwd
 
+def getNodePartitionKeys(df, client):
+    from dask import delayed
 
-def getNodePartitions(df, client):
-    df = df.persist()
     workers = client.scheduler_info()['workers']
+
     worker_partitions = {}
     for worker in workers:
         worker_partitions[worker] = []
 
     dask.distributed.wait(df)
     worker_part = client.who_has(df)
+
     for key in worker_part:
         if len(worker_part[key]) > 0:
             worker = worker_part[key][0]
-            partition = int(key[key.find(",") + 2:(len(key) - 1)])
-            worker_partitions[worker].append(partition)
+            worker_partitions[worker].append(key)
         else:
-            print("ERROR: In getNodePartitions, woker has no corresponding partition")
-
-    return dict((workers[worker]['name'], partitions) for worker, partitions in worker_partitions.items())
-
+            logging.error("ERROR: In getNodePartitionKeys, woker has no corresponding partition")
+            print("ERROR: In getNodePartitionKeys, woker has no corresponding partition")
```

Comment:
what if the worker doesnt have any because there are fewer partitions than workers?

Reply:
or because its not evently distributed

Reply:
hmmm, you are right, we dont need these error messages actually. If a worker does not have any partitions, its appropriatelly handles elsewhere. I have tested that

Reply:
ok